src/init.cpp

```diff
@@ -1024,6 +1025,13 @@ bool AppInitParameterInteraction(const ArgsManager& args)
         }
     }
 
+    if (args.IsArgSet("-maxcoinbaseweight")) {
+        const auto max_coinbase_weight = args.GetIntArg("-maxcoinbaseweight", node::BlockCreateOptions{}.coinbase_max_additional_weight);
+        if (max_coinbase_weight > MAX_BLOCK_WEIGHT) {
```

Comment:
a45af6af4e7a59cd14ef6e16b8e5e180fa9d564d

Should this test against `args.GetIntArg("-blockmaxweight", MAX_BLOCK_WEIGHT)` ? Or, what happens if a user sets `-blockmaxweight=1000 -maxcoinbaseweight=4000` ?

Reply:
Current `ClampOptions` has a comment that specifically permits `-maxcoinbaseweight` to exceed `-blockmaxweight`. It just means that all blocks will be empty. This seems like an unrealistic scenario anyway.